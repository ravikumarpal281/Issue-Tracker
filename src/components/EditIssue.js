import React from "react";
import { useFormik } from "formik";
import * as Yup from "yup";
import { Button, Container, Form } from "react-bootstrap";
import axios from "axios";
import { useEffect, useState } from "react";

const EditIssue = (props) => {
  const [issueState, setstate] = useState({
    id: "",
    name: "",
    description: "",
    severity: "",
    status: "",
  });

  useEffect(() => {
    console.log("Props Id", props.match.params.id);
    axios
      .get(`http://localhost:30001/Issues/${props.match.params.id}`)
      .then((response) => setstate(response.data))
      .catch((error) => console.error(error.message));
  }, []);

  const formik = useFormik({
    enableReinitialize: true,
    initialValues: {
      Name: issueState.name,
      Description: issueState.description,
      Severity: issueState.severity,
      Status: issueState.status,
    },
    validationSchema: Yup.object({
      Name: Yup.string()
        .min(2, "Minimum two characters required")
        .max(20, "Max twenty characters can be used for Name")
        .required("Name is reuired"),
      Description: Yup.string()
        .min(2, "Minimum two characters required")
        .max(50, "Max fifty characters")
        .required("Issue Description is required"),
      Severity: Yup.string().required("Severity is required"),
      Status: Yup.string().required("Status is required"),
    }),
    onSubmit: (values) => {
      axios
        .patch(`http://localhost:30001/Issues/${props.match.params.id}`, {
          name: values.Name,
          description: values.Description,
          severity: values.Severity,
          status: values.Status,
        })
        .catch((err) => console.error(err.message));
      props.history.push({ pathname: "/" });
    },
  });
  return (
    <div>
      <Container>
        <h1>Edit Issue</h1>
        <Form onSubmit={formik.handleSubmit}>
          <Form.Group>
            <Form.Label>Name</Form.Label>
            <Form.Control
              type="text"
              name="Name"
              onChange={formik.handleChange}
              value={formik.values.Name}
            ></Form.Control>
            {formik.errors.Name && formik.touched.Name && (
              <p style={{ color: "red" }}>{formik.errors.Name}</p>
            )}
          </Form.Group>
          <Form.Group>
            <Form.Label>Description</Form.Label>
            <Form.Control
              type="text"
              name="Description"
              onChange={formik.handleChange}
              value={formik.values.Description}
            ></Form.Control>
            {formik.errors.Description && formik.touched.Description && (
              <p style={{ color: "red" }}>{formik.errors.Description}</p>
            )}
          </Form.Group>
          <Form.Group>
            <label>
              Severity :
              <select
                style={{ marginLeft: "5px" }}
                value={formik.values.Severity}
                name="Severity"
                onChange={formik.handleChange}
              >
                <option value="Critical">Critical</option>
                <option value="Major">Major</option>
                <option value="Minor">Minor</option>
              </select>
            </label>
          </Form.Group>
          <Form.Group>
            <label>
              Status :{" "}
              <input
                style={{ marginLeft: "3px" }}
                type="radio"
                name="Status"
                value="Open"
                onChange={formik.handleChange}
                checked={formik.values.Status === "Open"}
              ></input>
              <label style={{ marginLeft: "3px" }}>Open</label>
              <input
                style={{ marginLeft: "5px" }}
                type="radio"
                name="Status"
                value="In Progress"
                onChange={formik.handleChange}
                checked={formik.values.Status === "In Progress"}
              ></input>
              <label style={{ marginLeft: "3px" }}>In Progress</label>
              <input
                style={{ marginLeft: "5px" }}
                type="radio"
                name="Status"
                value="Closed"
                onChange={formik.handleChange}
                checked={formik.values.Status === "Closed"}
              ></input>
              <label style={{ marginLeft: "3px" }}>Closed</label>
            </label>
          </Form.Group>
          <Button type="submit">Submit</Button>{" "}
          <Button onClick={() => props.history.push({ pathname: "/" })}>
            Cancel
          </Button>
        </Form>
      </Container>
    </div>
  );
};

export default EditIssue;
