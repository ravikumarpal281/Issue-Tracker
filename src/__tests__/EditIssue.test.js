import React from "react";
import toJson from "enzyme-to-json";
import Enzyme, { shallow, render, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import EditIssue from "../components/EditIssue";
import { Spinner, Button, Form } from "react-bootstrap";

Enzyme.configure({ adapter: new Adapter() });

const props = {
  history: {},
  location: {},
  match: {
    isExact: true,
    params: { id: "1" },
    path: "/editIssue/:id",
    url: "/editIssue/1",
  },
};
it("Edit Issue Snap", () => {
  const wrapper = shallow(<EditIssue {...props} />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

it("Edit Issue Headers Test", () => {
  const wrapper = shallow(<EditIssue {...props} />);
  expect(wrapper.find("div").length).toEqual(1);
  expect(wrapper.find("h1").text()).toEqual("Edit Issue");
});

it("Edit Issue Labels Test", () => {
  const wrapper = shallow(<EditIssue {...props} />);
  expect(wrapper.find(Form.Label).length).toEqual(2);
});

it("Submit And Cancel Button Check", () => {
  const wrapper = shallow(<EditIssue {...props} />);
  expect(wrapper.find(Button).length).toEqual(2);
});

it("Submit Button Check", () => {
  const mockEdit = jest.fn();
  const wrapper = shallow(<Button onClick={mockEdit}>Submit</Button>);
  const editButton = wrapper.find({ children: "Submit" });
  expect(editButton).toBeTruthy();
  editButton.simulate("click");
  expect(mockEdit).toHaveBeenCalled();
});

it("Delete Button Check", () => {
  const mockEdit = jest.fn();
  const wrapper = shallow(<Button onClick={mockEdit}>Cancel</Button>);
  const editButton = wrapper.find({ children: "Cancel" });
  expect(editButton).toBeTruthy();
  editButton.simulate("click");
  expect(mockEdit).toHaveBeenCalled();
});
