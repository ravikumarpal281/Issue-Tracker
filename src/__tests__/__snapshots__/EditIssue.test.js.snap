// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit Issue Snap 1`] = `
<div>
  <Container
    fluid={false}
  >
    <h1>
      Edit Issue
    </h1>
    <Card
      body={false}
      className="sign-in-form"
    >
      <Form
        className="form-style"
        inline={false}
        onSubmit={[Function]}
      >
        <FormGroup>
          <FormLabel
            column={false}
            srOnly={false}
          >
            Name
          </FormLabel>
          <FormControl
            name="Name"
            onChange={[Function]}
            type="text"
            value=""
          />
        </FormGroup>
        <FormGroup>
          <FormLabel
            column={false}
            srOnly={false}
          >
            Description
          </FormLabel>
          <FormControl
            name="Description"
            onChange={[Function]}
            type="text"
            value=""
          />
        </FormGroup>
        <FormGroup>
          <label>
            Severity :
            <select
              name="Severity"
              onChange={[Function]}
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
              value=""
            >
              <option
                value="Critical"
              >
                Critical
              </option>
              <option
                value="Major"
              >
                Major
              </option>
              <option
                value="Minor"
              >
                Minor
              </option>
            </select>
          </label>
        </FormGroup>
        <FormGroup>
          <label>
            Status :
             
            <input
              checked={false}
              name="Status"
              onChange={[Function]}
              style={
                Object {
                  "marginLeft": "3px",
                }
              }
              type="radio"
              value="Open"
            />
            <label
              style={
                Object {
                  "marginLeft": "3px",
                }
              }
            >
              Open
            </label>
            <input
              checked={false}
              name="Status"
              onChange={[Function]}
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
              type="radio"
              value="In Progress"
            />
            <label
              style={
                Object {
                  "marginLeft": "3px",
                }
              }
            >
              In Progress
            </label>
            <input
              checked={false}
              name="Status"
              onChange={[Function]}
              style={
                Object {
                  "marginLeft": "5px",
                }
              }
              type="radio"
              value="Closed"
            />
            <label
              style={
                Object {
                  "marginLeft": "3px",
                }
              }
            >
              Closed
            </label>
          </label>
        </FormGroup>
        <Button
          active={false}
          disabled={false}
          type="submit"
          variant="primary"
        >
          Submit
        </Button>
         
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          variant="primary"
        >
          Cancel
        </Button>
      </Form>
    </Card>
  </Container>
</div>
`;
